packages/i18n/src/create-i18n.ts

```diff
@@ -389,6 +406,44 @@ export const createI18n = < TextDomain extends string >(
 		hooks.addAction( 'hookRemoved', 'core/i18n', onHookAddedOrRemoved );
 	}
 
+	const numberFormatI18n: I18n[ 'numberFormatI18n' ] = (
+		number,
+		decimals = 0,
+		domain = 'default' as TextDomain
+	) => {
+		const localeData = ( getLocaleData( domain as TextDomain )?.[
+			''
+		] as I18nDomainMetadata< TextDomain > ) ?? {
+			lang: 'en_US',
+			domain: 'default',
+			plural_forms: 'n === 1 ? 0 : 1',
+		};
+
+		const wpLocale = localeData.lang || 'en_US';
+		// Convert WordPress locale format to JavaScript locale format by replacing underscores with hyphens
+		let jsLocale = wpLocale.replace( /_/g, '-' );
+
+		// Validate the locale using Intl.getCanonicalLocales and fallback to en-US if invalid
+		if ( ! isValidLocale( jsLocale ) ) {
+			jsLocale = 'en-US';
+		}
+
+		/**
+		 * Range limit of `maximumFractionDigits` in Node.js is 0-20
+		 * Max limit of `maximumFractionDigits` in Browsers is 0-100 ( {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#maximumfractiondigits} )
+		 *
+		 * Hence we limit the decimals to 20 to ensure compatibility across environments.
+		 */
+		const validDecimals = Math.max( 0, Math.min( decimals, 20 ) );
```

Comment:
What's the motivation for this? Is it really our responsibility to clamp this value? Seems more like the consumer's job.

Reply:
The reason behind this was that during testing I found out that while Browser supported upto 100 decimals , Nodejs was limited to around 20 decimal points and anything above that it would throw an error saying invalid range.
Hence I thought of adding clamp to this to prevent this error.